pandas/tests/frame/methods/test_update.py

```diff
@@ -136,7 +136,8 @@ def test_update_from_non_df(self):
     def test_update_datetime_tz(self):
         # GH 25807
         result = DataFrame([pd.Timestamp("2019", tz="UTC")])
-        result.update(result)
+        with tm.assert_produces_warning(FutureWarning):
```

Comment:
i think this is wrong.  in DataFrame.update we use `self.loc[:, col] = ...` in a way that produces a warning that we probably don't want.  so we need to change what we're doing there, not catch a warning here

Reply:
Yeah, agreed. Forgot to post here. Hadn't had time to take a closer look yet though. We probably have a couple more of those cases

Reply:
Opened #48468 to revert the update warning